### PR TITLE
fix for Privilege Block issue

### PR DESCRIPTION
**Pull request by wijith7**
### Proposed changes in this pull request

This PR fix the issue [1] from Audit Data Publishers

[1] ERROR {org.wso2.carbon.user.mgt.UserRealmProxy} -  Error occurred while accessing Java Security Manager Privilege Block

### CODE REVIEW

File: components/org.wso2.carbon.identity.data.publisher.audit.common/src/main/java/org/wso2/carbon/identity/data/publisher/audit/common/AuditDataPublisherUtils.java [hunk 1]
```diff
@@ -57,7 +57,9 @@ private AuditDataPublisherUtils() {     // Prevent initializing
         if (isTenantDomainNotBlank(actionHolderTenantDomain)) {
             tenantDomain.add(userTenantDomain);
         }
-        return (String[]) tenantDomain.toArray();
+
+        return tenantDomain.toArray(new String[0]);
+
```
Comment (hasuniea) [inline components/org.wso2.carbon.identity.data.publisher.audit.common/src/main/java/org/wso2/carbon/identity/data/publisher/audit/common/AuditDataPublisherUtils.java]: Please remove unnecessary new lines.

File: components/org.wso2.carbon.identity.data.publisher.audit.idp.properties/pom.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@
     <parent>
         <groupId>org.wso2.carbon.identity.data.publisher.audit</groupId>
         <artifactId>identity-data-publisher-audit</artifactId>
-        <version>1.0.4-SNAPSHOT</version>
+        <version>1.0.3</version>
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.data.publisher.audit.idp.properties/pom.xml]: Shouldn't we keep this snapshot ?.  Only the release build will change this and update to the snapshot of the next version.

**Comment by wijith7** [comment]
this resolves #8

**Comment by wijith7** [comment]
this resolves #3769

**Comment by wijith7** [comment]
this resolves #3769

**Comment by Epic55** [comment]
I have the same problem. I get this error when i try to change passwords for users on API or IS. I use IS as KM. But i don't have files AuditDataPublisherUtils.java, pom.xml, .DS_Store. How to solve this problem?

**Comment by wijith7** [comment]
is it coming intermittently

**Comment by Epic55** [comment]
> is it coming intermittently ?

No, always

**Pull request merged**